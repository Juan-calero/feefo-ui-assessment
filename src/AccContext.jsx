import { createContext } from "react"

const AccContext = createContext()

export default AccContext
