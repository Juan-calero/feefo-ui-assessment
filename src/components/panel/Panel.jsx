import React, { useContext } from "react"
import styled from "styled-components"
import AccountContext from "../../AccContext"
import AccMetric from "./metrics/AccMetric"
import Sales from "./sales/Sales"

const StPanel = styled.div`
  box-shadow: 3px 3px 12px #eee;
  border-radius: 7px;
  background-color: white;
`
const StMetricPanels = styled.div`
  @media (max-width: 599px) {
    flex-direction: column;
    & div:last-child {
      border-radius: 0 0 7px 7px;
    }
  }
  @media (min-width: 600px) {
    display: flex;
    justify-content: space-between;
    & div:first-child {
      border-right: 1px solid #e7e7e7;
      border-radius: 0 0 0 7px;
    }
    & div:last-child {
      border-radius: 0 0 7px 0;
    }
  }
`

function Panel() {
  const {
    salesOverview: { uploads, successfulUploads, linesAttempted, linesSaved },
  } = useContext(AccountContext)

  const getRate = (attempts, successes) =>
    Math.floor((successes * 100) / attempts)

  return (
    <StPanel>
      <Sales
        uploadsAdded={successfulUploads}
        linesAdded={linesSaved}
        aria-label='Sales Panel containing sales overview'
      />
      <StMetricPanels>
        <AccMetric
          percent={getRate(uploads, successfulUploads)}
          text='upload success'
          aria-label='Percentage Panel containing successful uploads'
        />
        <AccMetric
          percent={getRate(linesAttempted, linesSaved)}
          text='lines saved'
          aria-label='Percentage Panel containing saved lines'
        />
      </StMetricPanels>
    </StPanel>
  )
}

export default Panel
