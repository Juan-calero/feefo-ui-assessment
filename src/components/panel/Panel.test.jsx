import { render } from "@testing-library/react"
import AccContext from "../../AccContext"
import Panel from "./Panel"

const accountData = {
  salesOverview: {
    uploads: 8,
    successfulUploads: 3,
    linesAttempted: 20,
    linesSaved: 4,
    lastUploadDate: 1605001226079,
  },
}

const panel = render(
  <AccContext.Provider value={accountData}>
    <Panel />
  </AccContext.Provider>
)

test("Renders Panel", () => expect(panel).toMatchSnapshot())
