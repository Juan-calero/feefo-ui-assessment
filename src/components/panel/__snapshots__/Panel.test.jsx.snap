// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt dRpRPI"
      >
        <div
          class="sc-gsTCUz jFjbsv"
        >
          <div
            class="sc-dlfnbm hWWmxF"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-upload fa-w-16 upload"
              data-icon="upload"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                fill="currentColor"
              />
            </svg>
            <h2>
              Sales
            </h2>
            <svg
              aria-hidden="false"
              aria-label="information circle icon"
              class="svg-inline--fa fa-info-circle fa-w-16 info"
              data-icon="info-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              tabindex="0"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="salesText"
          >
            You had 
            <strong>
              3
               uploads
            </strong>
             and
             
            <strong>
              4
            </strong>
             lines added.
          </p>
        </div>
        <div
          class="sc-eCssSg zHUZW"
        >
          <div
            class="sc-bdfBwQ fXWavX"
          >
            <p
              class="percent"
            >
              37
              %
            </p>
            <p
              class="label"
            >
              UPLOAD SUCCESS
            </p>
          </div>
          <div
            class="sc-bdfBwQ fXWavX"
          >
            <p
              class="percent"
            >
              20
              %
            </p>
            <p
              class="label"
            >
              LINES SAVED
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt dRpRPI"
    >
      <div
        class="sc-gsTCUz jFjbsv"
      >
        <div
          class="sc-dlfnbm hWWmxF"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload fa-w-16 upload"
            data-icon="upload"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <h2>
            Sales
          </h2>
          <svg
            aria-hidden="false"
            aria-label="information circle icon"
            class="svg-inline--fa fa-info-circle fa-w-16 info"
            data-icon="info-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            tabindex="0"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              fill="currentColor"
            />
          </svg>
        </div>
        <p
          class="salesText"
        >
          You had 
          <strong>
            3
             uploads
          </strong>
           and
           
          <strong>
            4
          </strong>
           lines added.
        </p>
      </div>
      <div
        class="sc-eCssSg zHUZW"
      >
        <div
          class="sc-bdfBwQ fXWavX"
        >
          <p
            class="percent"
          >
            37
            %
          </p>
          <p
            class="label"
          >
            UPLOAD SUCCESS
          </p>
        </div>
        <div
          class="sc-bdfBwQ fXWavX"
        >
          <p
            class="percent"
          >
            20
            %
          </p>
          <p
            class="label"
          >
            LINES SAVED
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
